src/plat/pc99/machine/dev/vga/io.c

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2024, DornerWorks
+ *
+ * SPDX-License-Identifier: GPL-2.0-only
+ */
+#include <config.h>
+#include <arch/kernel/boot_sys.h>
+#include <arch/model/statedata.h>
+#include <machine/io.h>
+#include <plat/machine/io.h>
+#include <drivers/uart.h>
+
+#define VGA_PHYS 0xb8000 // Text buffer address
+
+//black background, light grey text, no blinking
+#define VGA_COLOR 0x07
+#define CLEAR_SCREEN_CHAR 32
+static uint64_t vga_addr = 0;
+static uint64_t vga_size = 0;
+// assume 80x25 console
+#define MAX_X_POS 80
+#define MAX_Y_POS 25
+
+/* Name kept 'serial_init' since other code expects to call this */
+void serial_init(uint16_t port)
```

Comment:
Since this is used on x86 only and looking at the places in `src/arch/x86/kernel/cmdline.c`where this is called, there seems some benefit if we make this `kernel_putDebugChar_init()`. And sanitize the code there, that deals with the command line arguments and setup. Looking at `src/arch/x86/model/statedata.c` we have `x86KSconsolePort` and `x86KSdebugPort`, which now only seem to have a use when `src/plat/pc99/machine/dev/com/io.c` is used. So I wonder if these variables should move then also to make this fully modular, as this seem the counterpart of `vga_addr` and `vga_size` in the end.